core/src/main/java/org/elasticsearch/common/inject/SpawnShittyModules.java

```diff
@@ -20,9 +20,19 @@
 package org.elasticsearch.common.inject;
 
 /**
+ * This interface can be added to a Module to spawn sub modules. DO NOT USE.
  *
+ * This is fundamentally broken.
+ * <ul>
+ * <li>If you have a plugin with multiple modules, return all the modules at once.</li>
+ * <li>If you are trying to make the implementation of a module "pluggable", don't do it.
+ * This is not extendable because custom implementations (using onModule) cannot be
+ * registered before spawnModules() is called.</li>
+ * </ul>
+ * @deprecated Create modules directly
  */
-public interface SpawnModules {
+@Deprecated
```

Comment:
> I wholeheartedly agree with you. And to add to that, if removing the crappy api introduces another crappy thing (api/design/etc..), then you need to rethink the change and sometimes avoid it until you come up with a better alternative (and that's where I believe we're standing on this issue)

But its not removed here, only deprecated. You just have sour grapes because we are telling you your code fucking sucks. Thats fine, I'm -1 to push this fix *without* the deprecation so we will figure this shit out. Great because its a blocker issue.

And guess what: I won't back down. Ever.